frontend/src/shared/components/actions/Select.tsx

```diff
@@ -0,0 +1,147 @@
+import Arrow from '@components/assets/icons/Arrow';
+import styled from '@emotion/styled';
+import { useBoolean } from '@hooks/useBoolean';
+import {
+  ReactNode,
+  useContext,
+  createContext,
+  MouseEvent,
+  useEffect,
+} from 'react';
+
+type BarProps = {
+  selectedValue?: string;
+  label?: string;
+  placeholder?: string;
+  onChange: (option: { value: string; label: string }) => void;
+  children: ReactNode;
+};
+
+type OptionProps = {
+  value: string;
+  children: ReactNode;
+};
+
+type OptionContextType = (option: { value: string; label: string }) => void;
+
+const selectContext = createContext<OptionContextType | undefined>(undefined);
+
+function Option({ value, children }: OptionProps) {
+  const onChange = useContext(selectContext);
+  const clickOption = (e: MouseEvent<HTMLLIElement>) => {
+    onChange?.({ value: value, label: e.currentTarget.textContent as string });
+  };
+
+  return <S.Option onClick={clickOption}>{children}</S.Option>;
+}
+
+function Bar({
+  selectedValue,
+  label,
+  placeholder = '선택해주세요.',
+  onChange,
+  children,
+}: BarProps) {
+  const [open, , closeDropdown, toggleDropdown] = useBoolean(false);
+
+  const handleChange = (option: { value: string; label: string }) => {
+    onChange(option);
+    closeDropdown();
+  };
+
+  const handleKeyDown = (e: KeyboardEvent) => {
+    if (e.key === 'Escape') {
+      closeDropdown();
+    }
+  };
+
+  useEffect(() => {
+    if (open) {
+      window.addEventListener('click', closeDropdown);
+      window.addEventListener('keydown', handleKeyDown);
+    }
+
```

Comment:
그리고 스크린리더 접근성을 위해 키보드 조작으로도 option을 선택할 수 있도록 열어주는 것도 좋을 것 같아요! (당장 코드 반영해달라는 리뷰는 아닙니당! 참고 코드만 슬쩍 두고갈게욥)


```suggestion
switch (e.key) {
        case 'Escape':
          closeDropdown();
          break;
        case 'ArrowDown':
          setFocusedIndex((prev) => (prev < options.length - 1 ? prev + 1 : 0));
          e.preventDefault();
          break;
        case 'ArrowUp':
          setFocusedIndex((prev) => (prev > 0 ? prev - 1 : options.length - 1));
          e.preventDefault();
          break;
        case 'Enter':
        case ' ':
          if (focusedIndex >= 0 && open) {
            handleSelect(options[focusedIndex]);
          }
          e.preventDefault();
          break;
        default:
          break;
      }
    };
```

Reply:
오 좋습니다!!!
참고해서 다음에 한번 시도해볼게요😀